### DISCUSSION TITLE
Enable thread pool events

### DISCUSSION THREAD

**Discussion by rabbitstack** [microsoft/perfview#1962] [Q&A] [answered]
Hi,

As far as I could deduce by looking at the kernel event parser source code, `perfview` can parse [thread pool](https://github.com/microsoft/perfview/blob/afd33a906961167ba995a6d71f35de9ea6a2bbdd/src/TraceEvent/Parsers/KernelTraceEventParser.cs#L12185) events. Mysteriously, I'm not able to grok out the way to enable these events so they start to get published by the system logger provider. Could you kindly point out how to enable them? Is there a respective `ENABLE_TRACE_FLAG_` bitmask constant one could squash into `EnableFlags` member of the `TRACE_EVENT_PROPERTIES` structure?

Thanks in advance

**Comment by brianrob** [comment, marked as answer]
These events can be enabled by enabling their provider GUID, which is `c861d0e2-a2c1-4d36-9f9c-970bab943a12`.

Because PerfView contains this parser, you can find it in the Provider Browser and choose it there, or you can also specify `/Providers:Thread Pool::Verbose`.  I've confirmed that if I enable this provider, the events show up in the Events view:

<img width="200" alt="image" src="https://github.com/microsoft/perfview/assets/6210322/23cd360b-5f39-4091-b2e0-8bcf144b143f">

**Comment by rabbitstack** [comment]
Thanks for the follow up on this , @brianrob.
